dev/devicelab/bin/run.dart

```diff
@@ -66,12 +68,40 @@ Future<void> main(List<String> rawArgs) async {
       localEngineSrcPath: localEngineSrcPath,
     );
 
-    if (!result['success'])
-      exitCode = 1;
-
     print('Task result:');
     print(const JsonEncoder.withIndent('  ').convert(result));
     section('Finished task "$taskName"');
+
+    if (!result['success']) {
+      exitCode = 1;
+      if (args['exit']) {
+        return;
+      }
+    }
+  }
+}
+
+void addTasks({
+  List<ManifestTask> tasks,
+  ArgResults args,
+  List<String> taskNames,
+}) {
+  if (args.wasParsed('continue-from')) {
+    final int index = tasks.indexWhere((ManifestTask task) => task.name == args['continue-from']);
+    if (index == -1) {
+      throw 'Invalid task name "${args['continue-from']}"';
+    }
+    tasks.removeRange(0, index);
+  }
+  // Only start skipping if user specified a task to continue from
+  bool isQualifyingStage;
+  bool isQualifyingHost;
+  for (ManifestTask task in tasks) {
+    isQualifyingStage = !args.wasParsed('stage') || task.stage == args['stage'];
```

Comment:
I would lift the stage above the loop so we look it up once. If its not parsed it is null right?

Then:

```
final String stage = args['stage'];
for (ManifestTask task in tasks) {
  isQualifyingStage = task.stage == stage;
}

```

Reply:
it would have to be `isQualifyingStage = stage == null || task.stage == stage;` because we want this to return true if no stage option was passed.